tests/test_recipes.py

```diff
@@ -16,19 +16,38 @@ def test_recipe_registration(basicApp):
 def test_recipe_build(basicApp, tmpdir):
     basicApp.plugins.get("RecipePlugin")
     test_recipe = basicApp.recipes.get("test_recipe")
+    output_folder = str(tmpdir.mkdir("output"))
+    output_data = test_recipe.build(output_dir=output_folder, no_input=True, extra_context=None)
+    assert output_data == os.path.join(output_folder, "My Package")
+
+
+def test_recipe_build_missing_extra_context_parameter(basicApp, tmpdir):
+    basicApp.plugins.get("RecipePlugin")
+    test_recipe = basicApp.recipes.get("test_recipe")
 
     output_folder = str(tmpdir.mkdir("output"))
-    output_data = test_recipe.build(output_folder, no_input=True)
+    output_data = test_recipe.build(output_dir=output_folder, no_input=True)
 
     assert output_data == os.path.join(output_folder, "My Package")
 
 
+def test_recipe_build_extra_context_parameter_userdata(basicApp, tmpdir):
+    basicApp.plugins.get("RecipePlugin")
+    test_recipe = basicApp.recipes.get("test_recipe")
+
+    output_folder = str(tmpdir.mkdir("output"))
+    user_data = {'project_name': 'TestProject'}
+    output_data = test_recipe.build(output_dir=output_folder, no_input=True, extra_context=user_data)
+
+    assert output_data == os.path.join(output_folder, "TestProject")
+
+
 def test_recipe_jinja(basicApp, tmpdir):
     basicApp.plugins.get("RecipePlugin")
     test_recipe = basicApp.recipes.get("test_recipe")
 
     output_folder = str(tmpdir.mkdir("output"))
-    test_recipe.build(output_folder, no_input=True)
+    test_recipe.build(output_dir=output_folder, no_input=True, extra_context=None)
```

Comment:
"output_dir=" should not be needed.
Otherwise nearly all existing code using this function will not working without modification

Reply:
Also is  "extra_context=None" needed?
If set to None, I hope it must not be set.

Reply:
Corrected.